src/data/real/pi.lean

```diff
@@ -142,4 +142,121 @@ lemma pi_lt_3141593 : pi < 3.141593 := by pi_upper_bound [
   27720/19601, 56935/30813, 49359/25163, 258754/130003, 113599/56868, 1101994/551163,
   8671537/4336095, 3877807/1938940, 52483813/26242030, 56946167/28473117, 23798415/11899211]
 
+
+noncomputable theory
+open real filter
+open_locale classical big_operators topological_space
+local notation `|`x`|` := abs x
+local notation `π` := real.pi
+
+theorem mvt (f f':ℝ→ℝ) {a b c:ℝ} (h : a ≤ b) (hderiv : ∀ (x ∈ set.Icc a b), has_deriv_at f (f' x) x)
+  (hbound : ∀ (x ∈ set.Icc a b), |f' x| ≤ c) : |f b - f a| ≤ c * (b - a) :=
```

Comment:
This theorem should either go to `analysis/calculus/mean_value`, or disappear. Note that the proof can be written as
```
norm_image_sub_le_of_norm_deriv_le_segment' (λ x hx, (hderiv x hx).has_deriv_within_at)
  (λ x hx, hbound x (set.Ico_subset_Icc_self hx)) (set.right_mem_Icc.mpr h)
```
so it might be easier to supply these arguments to `norm_image_sub_le_of_norm_deriv_le_segment'` instead of adding another theorem.